labgrid/driver/rawnetworkinterfacedriver.py

```diff
@@ -74,8 +74,11 @@ def start_record(self, filename, *, count=None, timeout=None):
             cmd.append("--duration")
             cmd.append(str(timeout))
         cmd = self._wrap_command(cmd)
-        with open(filename, "wb") as outdata:
-            self._record_handle = subprocess.Popen(cmd, stdout=outdata, stderr=subprocess.PIPE)
+        if filename == "-":
+            self._record_handle = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
```

Comment:
That's not a good idea (IMHO). If you make `stderr=subprocess.PIPE`, then every consumer of the subprocess _must_ periodically read from the stderr pipe, otherwise if it fills up it will deadlock the process. Making it mandatory would make the code consuming this have to do that, which is annoying. Better would be to make it optional, although this still has problems because if you do `stdout=subprocess.PIPE, stderr=subprocess.PIPE`, a consumer would need to `poll()` on both stderr and stdout to prevent deadlock. Anyway, given how tricky this is to get correct, I think leaving it is the correct answer for now. If an error occurs, the tcpdump error output will be visible in the logs/stderr/etc. which is sufficient to debug the problem without a huge burden on the consuming code.

Reply:
My expectation was that tcpdump doesn't produce too much output on stderr, but maybe that's too naive. I'm okay with keeping it as is.

Reply:
I've now dropped the `stderr=subprocess.DEVNULL` leaving it to the default (`None`).